modules/core/src/main/java/org/apache/ignite/internal/processors/datastreamer/DataStreamerImpl.java

```diff
@@ -672,7 +704,84 @@ else if (autoFlushFreq == 0)
         KeyCacheObject key0 = cacheObjProc.toCacheKeyObject(cacheObjCtx, null, key, true);
         CacheObject val0 = cacheObjProc.toCacheObject(cacheObjCtx, val, true);
 
-        return addDataInternal(Collections.singleton(new DataStreamerEntry(key0, val0)));
+        IgniteCacheFutureImpl futPerThread = null;
+
+        enterBusy();
+
+        try {
+            long threadId = Thread.currentThread().getId();
```

Comment:
Variables should be declared as close to it's usage as possible. Please, move declaration closer to the usage.